cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java

```diff
@@ -1843,6 +1847,7 @@ private void dohandle(Callback[] callbacks) throws IOException, UnsupportedCallb
                     } else {
                         final String defaultName = ncb.getDefaultName();
                         if (defaultName != null) {
+                            this.username = defaultName;
```

Comment:
OK.  I'm not clear on the intended semantics of the code so a lot of this is guessing and making this shape match that shape.